### PR TITLE
Fix Stringer implementation of fingerprint processor

### LINKED ISSUE #35174: The fingerprint processor incorrectly includes a function reference in a json.Marshal call
The `fingerprint.String` function [here](https://github.com/elastic/beats/blob/0807bb8d4947a7be0c28896c927f3ed0aa5aae13/libbeat/processors/fingerprint/fingerprint.go#L89):

```go
func (p *fingerprint) String() string {
	json, _ := json.Marshal(p.config)
	return procName + "=" + string(json)
}
```

calls `json.Marshal` on a config struct that includes the field `Method hashMethod`, with the definition `type hashMethod func() hash.Hash`. Marshalling function references isn't supported and will probably behave in unexpected/undesirable ways. In `fingerprint/hash.go` there is a lookup table mapping strings to `hashMethod`s, and this string form is probably what is intended, so either the `Marshal` call should be changed or the `Config` struct should use a more marshal-friendly wrapper for its hash method.

### PR DESCRIPTION
**Pull request by zipperle**
## Proposed commit message

Fix Stringer implementation of fingerprint processor.
It incorrectly included a function reference in a json.Marshal call.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #35174

### CODE REVIEW

**Comment by pierrehilbert** [comment]
Hello @zipperle,
Could you please fix the linter issue?

**Comment by zipperle** [comment]
Hi @pierrehilbert, sure, I fixed the linter issues.

**Comment by pierrehilbert** [comment]
@zipperle did you sign our CLA?

**Comment by pierrehilbert** [comment]
Setting back to draft in the meantime.

**Comment by zipperle** [comment]
@pierrehilbert yes I have signed the CLA

File: libbeat/processors/fingerprint/fingerprint_test.go [hunk 1]
```diff
@@ -459,6 +461,18 @@ func TestIgnoreMissing(t *testing.T) {
...
+func TestProcessorStringer(t *testing.T) {
+	testConfig, err := config.NewConfigFrom(mapstr.M{
+		"fields":   []string{"field1"},
+		"encoding": "hex",
+		"method":   "md5",
+	})
+	assert.NoError(t, err)
+	p, err := New(testConfig)
+	assert.NoError(t, err)
+	assert.Equal(t, `fingerprint={"Method":"md5","Encoding":"hex","Fields":["field1"],"TargetField":"fingerprint","IgnoreMissing":false}`, fmt.Sprint(p))
```
Comment (ycombinator) [inline libbeat/processors/fingerprint/fingerprint_test.go:473]: Suggest using `require.*` instead of `assert.*`.

**Review by ycombinator — COMMENTED** [review]
Changes LGTM.  Thank you for fixing! 

Just made a minor suggestion about the test assertions, that's all.

**Review by ycombinator — APPROVED** [review]

**Comment by pierrehilbert** [comment]
Thx @zipperle we are now all good to be merged!
Your change will be in our next minor version release.

**Pull request merged**